src/components/DisciplineListItem.tsx

```diff
@@ -10,27 +16,66 @@ interface DisciplineListItemProps {
   hasBadge: boolean
   rightChildren?: ReactElement
   disabled?: boolean
+  showProgress?: boolean
 }
 
+const Icon = styled.Image`
+  width: ${wp('7%')}px;
+  height: ${wp('7%')}px;
+`
+
+const IconContainer = styled.View`
+  position: absolute;
+  top: ${wp('3.5%')}px;
+  left: ${wp('3.5%')}px;
+`
+
 const DisciplineListItem = ({
   item,
   onPress,
   hasBadge,
   rightChildren,
-  disabled = false
+  disabled = false,
+  showProgress = false
 }: DisciplineListItemProps): ReactElement | null => {
   const { numberOfChildren, title, icon, apiKey } = item
   const badgeLabel = hasBadge ? numberOfChildren.toString() : undefined
   // Description either contains the number of children and the type of children or just the type of children if the number is shown as badge
   const description = hasBadge ? childrenLabel(item) : childrenDescription(item)
 
+  const [progress, setProgress] = useState<number>(0)
+
+  useFocusEffect(() => {
+    if (showProgress) {
+      getProgress(item).then(setProgress).catch(reportError)
+    }
+  })
+
+  const iconWithProgress = (
+    <>
+      <Progress.Circle
+        progress={progress}
+        size={Math.round(wp('14%'))}
```

Comment:
Why do we need `Math.round` around this? Can it only handle integers?

Reply:
unfortunately yes. Behavior is weird for non-integer values.